yb-voyager/cmd/analyzeSchema.go

```diff
@@ -667,67 +666,89 @@ func processCollectedSql(fpath string, stmt *string, formattedStmt *string, objT
 	}
 
-	*formattedStmt = strings.TrimRight(*formattedStmt, "\n") //removing new line from end
+	formattedStmt = strings.TrimRight(formattedStmt, "\n") //removing new line from end
 
 	sqlInfo := sqlInfo{
 		objName:       objName,
-		stmt:          *stmt,
-		formattedStmt: *formattedStmt,
+		stmt:          stmt,
+		formattedStmt: formattedStmt,
 	}
-	*sqlInfoArr = append(*sqlInfoArr, sqlInfo)
-	(*stmt) = ""
-	(*formattedStmt) = ""
+	return sqlInfo
 }
 
 func createSqlStrInfoArray(path string, objType string) []sqlInfo {
 	log.Infof("Reading %s in dir %s", objType, path)
 
 	var sqlInfoArr []sqlInfo
+	reportNextSql := 0
+
+	file, err := os.ReadFile(path)
+	if err != nil {
+		utils.ErrExit("Error while reading %q: %s", path, err)
+	}
+
+	lines := strings.Split(string(file), "\n")
+	for i := 0; i < len(lines); i++ {
+		currLine := lines[i]
+		if len(currLine) == 0 {
+			continue
+		}
+
+		if strings.Contains(strings.TrimLeft(currLine, " "), "--") {
+			reportNextSql = invalidSqlComment(currLine)
+			continue
+		}
 
+		var stmt, formattedStmt string
+		if isStartOfCodeBlockSqlStmt(currLine) {
+			stmt, formattedStmt = collectSqlStmtContainingCode(lines, &i)
+		} else {
+			stmt, formattedStmt = collectSqlStmt(lines, &i)
+		}
+		sqlInfo := processCollectedSql(path, stmt, formattedStmt, objType, &reportNextSql)
+		sqlInfoArr = append(sqlInfoArr, sqlInfo)
+	}
+
+	return sqlInfoArr
+}
+
+// returns true when sql stmt is a CREATE statement for TRIGGER, FUNCTION, PROCEDURE
+func isStartOfCodeBlockSqlStmt(line string) bool {
+	reCreateProc, _ := getCreateObjRegex("PROCEDURE")
+	reCreateFunc, _ := getCreateObjRegex("FUNCTION")
+	reCreateTrigger, _ := getCreateObjRegex("TRIGGER")
+
+	return reCreateProc.MatchString(line) || reCreateFunc.MatchString(line) || reCreateTrigger.MatchString(line)
+}
+
+func collectSqlStmtContainingCode(lines []string, i *int) (string, string) {
 	// 0 -> code block is not started
 	// 1 -> code block is being traversed
 	// 2 -> code block is completed/terminated
 	dollarQuoteFlag := 0
 	// Delimiter to outermost Code Block if nested Code Blocks present
 	codeBlockDelimiter := ""
-	reportNextSql := 0
-
-	file, err := os.Open(path)
-	if err != nil {
-		utils.ErrExit("Error while opening %q: %s", path, err)
-	}
-	defer file.Close()
 
-	scanner := bufio.NewScanner(file)
 	stmt := ""
 	formattedStmt := ""
 
-	// assemble array of lines, each line ends with semicolon
-	for scanner.Scan() {
-		currLine := scanner.Text()
+	for ; *i < len(lines); *i++ {
+		currLine := lines[*i]
 		if len(currLine) == 0 {
 			continue
 		}
 
-		if strings.Contains(currLine, "--") { //in case there is a space before '--'
-			reportNextSql = invalidSqlComment(currLine)
-			if dollarQuoteFlag == 0 { // ignore comment only if it is outside a DDL
-				continue
-			}
-		}
-
 		stmt += currLine + " "
 		formattedStmt += currLine + "\n"
 
 		// Assuming that both the dollar quote strings will not be in same line
 		if dollarQuoteFlag == 0 {
```

Comment:
Use switch-case?

Please, use named constants for 0, 1, and 2. Without that the code is difficult to read.